### ISSUE TITLE
LocalStore.list_prefix adds directory prefix to all paths

### ISSUE THREAD

**Issue by dcherian** [zarr-developers/zarr-python#2190]
### Zarr version

v3

### Numcodecs version

?

### Python Version

?

### Operating System

?

### Installation

?

### Description

`list_prefix("")` for a LocalStore seems inconsistent with the other stores. Discovered in #2189

```
'/private/var/folders/sc/hsdthq2x7mnbfvwnk7rp4xmr0000gn/T/pytest-of-deepak/pytest-854/test_zarr_hierarchy_local_0/zarr.json'
```

Also:
https://github.com/zarr-developers/zarr-python/blob/b1ecdd544823404faf37e3111e868261178aa3e5/tests/v3/test_store/test_local.py#L39-L40
LOL

### Steps to reproduce

```python
```

**Comment by zoj613** [comment]
In my very inexperienced eyes, this looks like the cause:
https://github.com/zarr-developers/zarr-python/blob/b1ecdd544823404faf37e3111e868261178aa3e5/src/zarr/store/local.py#L204-L211

The root directory path is not stripped away and returned as is (line 206). I am not sure why the code appears to be duplicated in that function? Maybe someone who was implementing the correct code (lines 208 - 211) forgot to delete the previous block?

**Comment by jhamman** [comment]
I think @d-v-b is working on a related set of fixes here: https://github.com/zarr-developers/zarr-python/pull/2064

**Comment by d-v-b** [comment]
yep, and I can split the fixes out of #2064 into their own PR

**Comment by d-v-b** [comment]
#2064 is in, so I'm curious to see if the problem you observed is still around. 

https://github.com/zarr-developers/zarr-python/blob/b1ecdd544823404faf37e3111e868261178aa3e5/tests/v3/test_store/test_local.py#L39-L40

LOL indeed, this bit me when working on the PR because I had no idea why my changes to the localstore implementation were not making tests fail :)
